src/core/types/keys/key_agreement/public_key.rs

```diff
@@ -0,0 +1,243 @@
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::from_hex(s.to_owned())
+    }
+}
+
+impl TryFrom<Vec<u8>> for KeyAgreementPublicKey {
+    type Error = CommonError;
+
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        value.as_slice().try_into()
+    }
+}
+
+impl TryFrom<&[u8]> for KeyAgreementPublicKey {
+    type Error = crate::CommonError;
+
+    fn try_from(slice: &[u8]) -> Result<Self> {
+        X25519PublicKey::try_from_slice(slice)
+            .map_err(|_| CommonError::InvalidKeyAgreementPublicKeyFromBytes {
+                bad_value: slice.to_vec().into(),
+            })
+            .map(|k| k.into())
+    }
+}
+
+impl KeyAgreementPublicKey {
+    pub fn from_hex(hex: String) -> Result<Self> {
+        Exactly32Bytes::from_str(hex.as_str())
+            .map_err(|_| CommonError::InvalidKeyAgreementPublicKeyFromHex {
+                bad_value: hex,
+            })
+            .and_then(|b| b.to_vec().try_into())
+    }
+
+    pub fn to_hex(&self) -> String {
+        hex_encode(self.to_bytes())
+    }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        self.secret_magic.to_bytes().to_vec()
+    }
+}
+
+impl HasSampleValues for KeyAgreementPublicKey {
+    fn sample() -> Self {
+        Self::sample_alice()
+    }
+
+    fn sample_other() -> Self {
+        Self::sample_bob()
+    }
+}
+
+impl KeyAgreementPublicKey {
+    /// A sample used to facilitate unit tests.
+    ///
+    /// `8679bc1fe3210b2ce84793668b05218fdc4c220bc05387b7d2ac0d4c7b7c5d10`
+    pub fn sample_alice() -> Self {
+        Self::from_hex(
+            "8679bc1fe3210b2ce84793668b05218fdc4c220bc05387b7d2ac0d4c7b7c5d10"
+                .to_owned(),
+        )
+        .unwrap()
+    }
+
+    /// A sample used to facilitate unit tests.
+    ///
+    /// `c0f0d9d1b1f9c8c9f9b9d1f0c9f0c8c9f9b9d1f0c9f0c8c9f9b9d1f0c9f0c8c9`
+    pub fn sample_bob() -> Self {
+        Self::from_hex(
+            "c0f0d9d1b1f9c8c9f9b9d1f0c9f0c8c9f9b9d1f0c9f0c8c9f9b9d1f0c9f0c8c9"
+                .to_owned(),
+        )
+        .unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[allow(clippy::upper_case_acronyms)]
+    type SUT = KeyAgreementPublicKey;
+
+    #[test]
+    fn equality() {
+        assert_eq!(SUT::sample(), SUT::sample());
+        assert_eq!(SUT::sample_other(), SUT::sample_other());
+    }
+
+    #[test]
+    fn inequality() {
+        assert_ne!(SUT::sample(), SUT::sample_other());
+    }
+
+    #[test]
+    fn from_hex() {
+        assert_eq!(
+            SUT::from_hex("8679bc1fe3210b2ce84793668b05218fdc4c220bc05387b7d2ac0d4c7b7c5d10".to_owned()),
+            Ok(SUT::sample())
+        );
+    }
+
+    #[test]
+    fn to_hex() {
+        assert_eq!(
+            SUT::sample().to_hex(),
+            "8679bc1fe3210b2ce84793668b05218fdc4c220bc05387b7d2ac0d4c7b7c5d10"
+        );
+    }
+
+    #[test]
+    fn to_bytes() {
+        assert_eq!(
+            SUT::sample().to_bytes(),
+            vec![
```

Comment:
prefer using hex, we really dont like arrays of base10 representation of bytes. They are not so portable, also longer.. So you can:

```rust
assert_eq!(hex_encode(SUT::sample().to_bytes()), "...");
```